.github/workflows/upgrade.yaml

```diff
@@ -0,0 +1,57 @@
+name: Update vCluster dep
+on:
+  workflow_dispatch:
+    inputs:
+      releaseTag:
+        description: 'Release tag in form vX.Y.Z'
+        required: true
+        type: string
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: 'true'
+
+      - name: Configure git
+        run: |
+          git config --global url.https://"$GH_ACCESS_TOKEN"@github.com/.insteadOf https://github.com/
+          # set git info
+          git config --global user.name "Loft Bot"
+          git config --global user.email 'loft-bot@users.noreply.github.com'
+        env:
+          GH_ACCESS_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Update vcluster in vcluster-sdk
+        env:
+          GH_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
+          RELEASE_TAG: ${{ inputs.releaseTag }}
+        run: |
+          go mod tidy
+          go mod vendor
+
+          go run ./hack/bump-vcluster-dep.go
+
+          branch_name="upgrade-vcluster-to-${RELEASE_TAG}"
+          git switch -c ${branch_name}
+
+          git add --all
+
+          # if there are no changes, exit early
+          if git diff-index --quiet HEAD --; then
+            exit 0
+          fi
+          echo "Changes detected"
+
+          # create a PR in vcluster-sdk
+          git commit -m "chore: bump vCluster dependency to ${RELEASE_TAG}"
+          git push -u origin -f ${branch_name}
+          gh pr create --fill --head ${branch_name}
```

Comment:
Maybe something a bit more descriptive in the PR body, the commit message might not always be fitting here.

```yaml
gh pr create \
  --title "chore: bump vCluster dependency to ${RELEASE_TAG}" \
  --body "This PR updates the vCluster dependency to version ${RELEASE_TAG}.
```

Reply:
changed :+1: